node/src/node_configurator/configurator.rs

```diff
@@ -199,79 +198,24 @@ impl Configurator {
-                     not present despite mnemonic seed in place!"
-                )
-            },
-        ) {
-            Ok(val) => val,
-            Err(e) => return Err(e),
-        };
-
-        let consuming_wallet_address =
-            match Self::recalculate_consuming_wallet(mnemonic, derivation_path) {
-                Ok(wallet) => wallet.string_address_from_keypair(),
-                Err(e) => return Err((KEY_PAIR_CONSTRUCTION_ERROR, e)),
-            };
-
-        let earning_wallet_address = match Self::process_value_with_specific_none(
-            self.persistent_config.earning_wallet_address(),
-            || {
-                panic!(
-                    "Database corrupted: missing earning wallet address \
-                      despite other values for wallets in place!"
-                )
-            },
-        ) {
-            Ok(val) => val,
-            Err(e) => return Err(e),
-        };
-
-        Ok((consuming_wallet_address, earning_wallet_address))
-    }
-
-    fn process_value_with_specific_none<T>(
-        data: Result<Option<T>, PersistentConfigError>,
-        closure_for_none: fn() -> (u64, String),
-    ) -> Result<T, (u64, String)> {
-        match data {
-            Ok(value) => match value {
-                None => Err(closure_for_none()),
-                Some(value) => Ok(value),
-            },
-            Err(e) => Err((CONFIGURATOR_READ_ERROR, format!("{:?}", e))),
-        }
-    }
-
-    fn recalculate_consuming_wallet(
-        seed: PlainData,
-        derivation_path: String,
-    ) -> Result<Wallet, String> {
-        match Bip32ECKeyProvider::try_from((seed.as_ref(), derivation_path.as_str())) {
-            Err(e) => Err(format!(
-                "Consuming wallet address error during generation: {}",
-                e
+        let consuming_wallet_opt_result = self.persistent_config.consuming_wallet(&db_password);
+        let earning_wallet_opt_result = self.persistent_config.earning_wallet();
+        match (consuming_wallet_opt_result, earning_wallet_opt_result) {
+            (Ok(None), Ok(None)) => {
+                Err((MISSING_DATA, "Wallet pair not yet configured".to_string()))
+            }
+            (Ok(Some(consuming_wallet)), Ok(Some(earning_wallet))) => Ok((
+                format!("{:?}", consuming_wallet.address()),
+                format!("{:?}", earning_wallet.address()),
             )),
-            Ok(kp) => Ok(Wallet::from(kp)),
+            (Ok(None), Ok(Some(_))) => {
+                panic!("Database corrupted: earning wallet exists but consuming wallet does not")
+            }
+            (Ok(Some(_)), Ok(None)) => {
+                panic!("Database corrupted: consuming wallet exists but earning wallet does not")
+            }
+            (Err(ce), _) => Err((CONFIGURATOR_READ_ERROR, format!("{:?}", ce))),
+            (_, Err(ee)) => Err((CONFIGURATOR_READ_ERROR, format!("{:?}", ee))),
```

Comment:
Don't you want to add a guidance of what wallet the error is attached to within your message?